src/runtimes/index.js

```diff
@@ -1,8 +1,82 @@
+const { extname, basename } = require('path')
+
 const goRuntime = require('./go')
 const jsRuntime = require('./node')
 const rustRuntime = require('./rust')
 
+/**
+ * Finds functions for a list of paths using a specific runtime. The return
+ * value is an object containing an array of the functions found (`functions`)
+ * and an array with the paths that haven't been recognized by the runtime
+ * (`remainingPaths`).
+ *
+ * @param   {Array<String>} paths
+ * @param   {Object} runtime
+ * @returns {Promise<Object>}
+ */
+const findFunctionsInRuntime = async function ({ dedupe = false, paths, runtime }) {
```

Comment:
The `dedupe` parameter dictates whether `functions/test.js` and `functions/test/index.js` will yield two entries or merged into one (according to the precedence rules). I've added this because methods like `listFunctions` and `listFunctionFiles` are still returning all the functions and corresponding files regardless of naming conflicts simply to maintain backwards compatibility, but this feels like something we should revisit.